security/tls-1.3.md

```diff
@@ -0,0 +1,213 @@
+- kube-system (Kubernetes)
+- Other (metal3-dev-env)
+
+## Background information
+
+### TLS 1.3 support in Golang
+
+Starting in Go 1.13, TLS 1.3 support has been available. In Go 1.13, TLS 1.3 was
+"opt-in", which led many projects adding `GODEBUG=tls13=1` to their
+environments. This is only necessary and meaningful in Go 1.13. In later
+versions TLS 1.3 is "opt-out", ie. you need to set TLS minimum and maximum
+versions in project's `tls.Config`.
+
+### TLS 1.0 and TLS 1.1 in Golang
+
+Starting in Go 1.18, TLS 1.0 and TLS 1.1 have been disabled by default in
+Golang. This leaves TLS 1.2 and TLS 1.3 as the only versions of TLS in Go 1.18
+and later, unless project explicitly adds support for TLS 1.0 and TLS 1.1 back
+via `tls.Config`.
+
+### Configuration of min and max TLS versions
+
+Many of the projects have only recently or not at all added support for
+configuring minimum or maximum TLS versions. Without project's support for
+configuration, TLS version negotiation is left to Golang and the client.
+
+## Components
+
+### BMO
+
+**controller-manager**:
+
+- Port `8443`: TLS 1.2, TLS 1.3
+
+**ironic**:
+
+For Ironic, it should be noted that:
+
+- Ironic external endpoints are secured, while pod-internal traffic is not.
+  `httpd` (Apache) handles TLS termination.
+- IPA image serving from `httpd` port `6180` might be limited to insecure due
+  PXE limitations (to be verified)
+- Node image server is deployment specific. IPA client accessing the image
+  server is limited to TLS 1.2 (see Oslo below)
+
+Ports:
+
+- Port `5049`: HTTP (Ironic Inspector API)
+- Port `5050`: TLS 1.2, TLS 1.3 (httpd - Inspector endpoint)
+- Port `6180`: HTTP (httpd - serving IPA images)
+- Port `6385`: TLS 1.2, TLS 1.3 (httpd - Ironic endpoint)
+- Port `6388`: HTTP (Ironic API)
+
+Ironic endpoints support setting minimum and maximum TLS versions.
+
+More info:
+
+- [OSLO library does not support](https://docs.openstack.org/oslo.service/latest/configuration/index.html#ssl.version)
+  TLS newer than 1.2 yet, TLS 1.3 support is planned
+- [IPA TLS configuration documentation](https://docs.openstack.org//ironic-python-agent/latest/doc-ironic-python-agent.pdf)
+
+### CAPI
+
+All CAPI controllers support setting minimum and maximum TLS versions, with the
+exception of `CAPD` which is a test provider.
+
+**capi-kubeadm-bootstrap**:
+
+- Port `9443`: TLS 1.2, TLS 1.3
+- Port `9440`: HTTP (healtz)
+- Port `8080`: HTTP (metrics)
+
+**capi-kubeadm-control-plane**:
+
+- Port `9443`: TLS 1.2, TLS 1.3
+- Port `9440`: HTTP (healtz)
+- Port `8080`: HTTP (metrics)
+
+**capi-controller-manager**:
+
+- Port `9443`: TLS 1.2, TLS 1.3
+- Port `9440`: HTTP (healtz)
+- Port `8080`: HTTP (metrics)
+
+### CAPM3
+
+**capm3-controller-manager**:
+
+- Port `9443`: TLS 1.2, TLS 1.3
+- Port `9440`: HTTP (healtz)
+- Port `8080`: HTTP (metrics)
+
+**ipam-controller-manager**:
+
+- Port `9443`: TLS 1.2, TLS 1.3
+- Port `9440`: HTTP (healtz)
+- Port `8080`: HTTP (metrics)
```

Comment:
1.3 can't be configured ATM, also BMO is not mentioned in this list at all

Reply:
I have only mentioned specifically if it CAN be configured. Maybe I'll add a comment if it also can't for clarity.

BMO is on the top with Ironic.

Reply:
Done.